pkg/kubelet/secret/secret_manager.go

```diff
@@ -209,8 +209,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 2 minutes (which is a lot, but...)
+	ttl := 2 * time.Minute
```

Comment:
I though about it and started also some internal thread about it. But my new proposal is as following:
- we will introduce a new annotation "secret-ttl' (please suggest better name for it) that we will be attaching to secrets
- we will write a very simple controller that will be watching nodes and secrets and based on cluster size will be setting the value of "secret-ttl" annotation for all secrets (also will be doing periodic resyncs to check if the value is up-to-date)
- we will introduce a minor modifcation to secret manager so that it will be first trying to take a ttl from the secret annotation, and only in case this doesn't exist, it will be using the default 1 minute ttl

Advantages of this approach:
- this won't introduce any changes to how we setup kubernetes (i.e. we don't introduce/change any kubelet flags, etc.)
- all the changes will be in pretty much two places: minor changes in secret manager and a new 'secret controller'
- once we have a proper long term solution, reverting those will be super trivial

And the one that I think is the most important is that we will be able to decrease this 1minute ttl in small clusters even to 0 (which mean always fetch). So this value will be higher only for very large clusters (probably bigger than 2k nodes or sth like that).

WDYT?
@thockin what was also interested in this issue